applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_local_stress_response_function.cpp

```diff
@@ -430,90 +424,90 @@ namespace Kratos
+                            DofsVectorType &rElementalDofList, const Array1DComponentsPointerType& rTracedDof) const
     {
         KRATOS_TRY;
 
-        if(rResult.size() != rElementalDofList.size())
+        if(rResult.size() != rElementalDofList.size()) {
             rResult.resize(rElementalDofList.size());
+        }
 
         const unsigned int num_GP = mpTracedElement->GetGeometry().IntegrationPointsNumber(mpTracedElement->GetIntegrationMethod());
         const double prefactor = 1.0 / (1.0 + num_GP);
 
         const IndexType id_node_1 = mpTracedElement->GetGeometry()[0].Id();
         const IndexType id_node_2 = mpTracedElement->GetGeometry()[1].Id();
 
-        for(IndexType i = 0; i < rElementalDofList.size(); ++i)
-        {
-            if (rElementalDofList[i]->GetVariable().Name() == rDofLabel)
-            {
-                if (rElementalDofList[i]->Id() == id_node_1)
+        for(IndexType i = 0; i < rElementalDofList.size(); ++i) {
+            if (rElementalDofList[i]->GetVariable() == *rTracedDof) {
+                if (rElementalDofList[i]->Id() == id_node_1) {
                     rResult[i] = prefactor * (num_GP + 1 - mIdOfLocation);
-                else if (rElementalDofList[i]->Id() == id_node_2)
+                } else if (rElementalDofList[i]->Id() == id_node_2) {
                     rResult[i] = -prefactor * mIdOfLocation;
+                }
             }
         }
 
         KRATOS_CATCH("");
     }
 
     void AdjointLocalStressResponseFunction::CalculateNodeParticularSolutionLinearElement2N(Vector& rResult,
-                     DofsVectorType &rElementalDofList, const std::string& rDofLabel) const
+                     DofsVectorType &rElementalDofList, const Array1DComponentsPointerType& rTracedDof) const
     {
         KRATOS_TRY;
 
-        if(rResult.size() != rElementalDofList.size())
+        if(rResult.size() != rElementalDofList.size()) {
             rResult.resize(rElementalDofList.size());
+        }
 
-        for(IndexType i = 0; i < rElementalDofList.size(); ++i)
-        {
+        for(IndexType i = 0; i < rElementalDofList.size(); ++i) {
             if (rElementalDofList[i]->Id() == mpTracedElement->GetGeometry()[mIdOfLocation-1].Id() &&
-                rElementalDofList[i]->GetVariable().Name() == rDofLabel)
-            {
-                if (mIdOfLocation == 1)
+                rElementalDofList[i]->GetVariable() == *rTracedDof) {
+                if (mIdOfLocation == 1) {
                     rResult[i] = 1.0;
-                else if (mIdOfLocation == 2)
+                } else if (mIdOfLocation == 2) {
                     rResult[i] = -1.0;
+                }
             }
         }
 
         KRATOS_CATCH("");
     }
 
-    void AdjointLocalStressResponseFunction::FindCorrespondingDofLabel(std::string& rDofLabel) const
+    void AdjointLocalStressResponseFunction::FindVariableComponent(Array1DComponentsPointerType& rTracedDof) const
```

Comment:
Ref of a pointer?

Reply:
this was also new to me. @philbucher suggested doing so.

Reply:
Ok got it, but i think it is only necessary to pass it by reference in the FindVariableComponent function, the otheres are just using it and it could be passed by value there.

Reply:
yes for the other fct it should be sufficent to only pass the pointer (without reference), since there you don't want to change the pointer (only use ti)

Reply:
thanks for the hint. I pass now the directly the pointer for the other fcts.